lib/ui/Editor.js

```diff
@@ -621,6 +626,14 @@ Editor.prototype._updateCursor = function () {
     self.screen.program.hideCursor();
   }
 };
+Editor.prototype._clipCursorToContent = function (cursor, content) {
```

Comment:
Ahh.. you should be able to use `textBuf.getLastRow` for `y` and `textBuf.lineLengthForRow` for `x` in that case. (`getEndPosition` source [here](https://github.com/atom/text-buffer/blob/master/src/text-buffer.coffee#L1081-L1083) if you're interested)